src/anndata/_io/__init__.py

```diff
@@ -1,40 +1,58 @@
 from __future__ import annotations
 
-from .h5ad import read_h5ad, write_h5ad
-from .read import (
-    read_csv,
-    read_excel,
-    read_hdf,
-    read_loom,
-    read_mtx,
-    read_text,
-    read_umi_tools,
-    read_zarr,
-)
-from .write import write_csvs, write_loom
-
-
-def write_zarr(*args, **kw):
-    from .zarr import write_zarr
-
-    return write_zarr(*args, **kw)
-
-
-# We use this in test by attribute access
-from . import specs  # noqa: F401, E402
-
-__all__ = [
-    "read_csv",
-    "read_excel",
-    "read_h5ad",
-    "read_hdf",
-    "read_loom",
-    "read_mtx",
-    "read_text",
-    "read_umi_tools",
-    "read_zarr",
-    "write_csvs",
-    "write_h5ad",
-    "write_loom",
-    "write_zarr",
-]
+import warnings
+
+__all__: list[str] = []
+
+
+def __getattr__(key: str):
+    from .._core.sparse_dataset import sparse_dataset
+    from .h5ad import read_h5ad, write_h5ad
+    from .read import (
+        read_csv,
+        read_excel,
+        read_hdf,
+        read_loom,
+        read_mtx,
+        read_text,
+        read_umi_tools,
+        read_zarr,
+    )
+    from .specs import read_elem, write_elem
+    from .write import write_csvs, write_loom
+
+    def write_zarr(*args, **kw):
+        from .zarr import write_zarr
+
+        return write_zarr(*args, **kw)
+
+    # We use this in test by attribute access
+    from . import specs  # noqa: F401, E402
```

Comment:
Any reason why we shouldn’t just stop doing that?

Reply:
Would need to investigate what it's doing there or what that comment even means.  Maybe an issue?

Reply:
Well, I deleted it and nothing happened so....

Reply:
I put that import and comment there because Isaac made the tests do

```py
import anndata as ad

do_stuff_with(ad._io.specs.foobar)
```

instead of importing that module. It happened to work because some other code somewhere else caused `anndata._io.specs` to be imported, but “happens to work because of side effects” is not what I want to build upon.

Now that the tests no longer access stuff that way, this import is no longer necessary.